src/Sulu/Component/Content/Mapper/ContentMapper.php

```diff
@@ -1640,13 +1640,13 @@ private function getLocalizedUrlsForPage(Page $page, NodeInterface $node, $websp
         $contentType = $this->contentTypeManager->get($property->getContentTypeName());
 
         foreach ($webspace->getAllLocalizations() as $localization) {
-
             // prepare translation vars
             $locale = $localization->getLocalization();
+            $shadowLocale = $this->getShadowLocale($node, $localization->getLocalization());
```

Comment:
@danrot this function returns the shadow locale or the second parameter ... so if the page is a shadow it returns the shadow locale else the requested locale